### PR TITLE
Added endpoint /api/auth/session-ttl

### PR DESCRIPTION
**Pull request by Wesley-Lau**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,4 +5,5 @@ sessions/
 logs/
 yaml/
 newrelic_agent.log
-.env
\ No newline at end of file
+.env
+*localtest*
```
Comment (knockknockyoo) [inline .gitignore]: This is unnecessary for this pr.

File: services/ttl.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+const {MongoDBCollection} = require("../crdc-datahub-database-drivers/mongodb-collection");
+const {DATABASE_NAME, SESSION_COLLECTION} = require("../crdc-datahub-database-drivers/database-constants");
+const dbConnector = new DatabaseConnector(config.mongo_db_connection_string);
+
+
+
+
+const getTTL = (req, res) => {
+    const sessionID = req.sessionID
+    if (sessionID){
```
Comment (knockknockyoo) [inline services/ttl.js]: Why did you declare sessionID?

File: services/ttl.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+const config = require('../config');
+const {DatabaseConnector} = require("../crdc-datahub-database-drivers/database-connector");
+const {MongoDBCollection} = require("../crdc-datahub-database-drivers/mongodb-collection");
+const {DATABASE_NAME, SESSION_COLLECTION} = require("../crdc-datahub-database-drivers/database-constants");
+const dbConnector = new DatabaseConnector(config.mongo_db_connection_string);
+
```
Comment (knockknockyoo) [inline services/ttl.js]: Please, remove all unused imports.

File: services/ttl.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+const getTTL = (req, res) => {
+    const sessionID = req.sessionID
+    if (sessionID){
+        let expires = req.session.cookie.expires;
+        if (!expires){
+            response = {ttl: 0}
+        }
+        else {
+            let dt = new Date(expires);
```
Comment (knockknockyoo) [inline services/ttl.js]: Why did you use let? Do you understand the difference between let and const?

https://www.freecodecamp.org/news/var-let-and-const-whats-the-difference/

**Review by knockknockyoo — CHANGES_REQUESTED** [review]
This changes must be addressed. Thanks.

File: services/ttl.js (hunk 2 above)
Comment (Wesley-Lau) [reply services/ttl.js]: I just though I might check if there is sessions before going through.

Comment (knockknockyoo) [reply services/ttl.js]: You should not declare another variable. I dont see sessionID reused in this function.

File: .gitignore [hunk 5]
```diff
@@ -5,4 +5,3 @@ sessions/
 logs/
 yaml/
 newrelic_agent.log
-.env
```
Comment (knockknockyoo) [inline .gitignore:8]: Please, include this

File: services/ttl.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+            let ttl = Math.round((dt.valueOf() - Date.now()));
+            response = {ttl: ttl};
+        }
+        res.json(response)
```
Comment (knockknockyoo) [inline services/ttl.js]: Look at response. This is always undefined.

**Review by AustinSMueller — CHANGES_REQUESTED** [review]
The service currently runs into an error when starting, the file ttl-query was removed in the latest commit but another file is trying to import this. Please correct this and re-test then submit for review again. Thank you

**Pull request closed without merging**